Squeezer/src/main/java/uk/org/ngo/squeezer/service/SqueezeService.java

```diff
@@ -1416,6 +1467,35 @@ public void downloadItem(JiveItem item) throws HandshakeNotCompleteException {
             mDelegate.requestItems(-1, callback).params(command.params).cmd(command.cmd()).exec();
         }
 
+        public void randomPlayFolder(JiveItem item) throws HandshakeNotCompleteException {
+
+            // Has to be instantiated once to have an mDelegate
+            new RandomPlayDelegate(mDelegate);
+
+            SlimCommand command = item.randomPlayFolderCommand();
+            String folderID = "";
+            try {
+                folderID = (String) command.params.get("folder_id");
+            } catch (Exception e) {
+                Log.e(TAG, "randomPlayFolder: Could not cast value of 'folder_id' to String");
```

Comment:
Decide what to do if there is no folder id.
Currently it will "loadRandomPlayed" with an empty folder id.
Maybe just skip the try/catch or alternatively leave the function (this will silently ignore the request so it is not a good user experience).
Also remove the unused "throws HandshakeNotCompleteException".

Reply:
I added messages to the user for successfull and erroneous Random Play. The function is left with an additional log error message in case of a not found folder id. I removed the throws.